packages/dom/src/document.rs

```diff
@@ -697,12 +692,22 @@ impl Document {
             height: AvailableSpace::Definite(size.height.to_f32_px()),
         };
 
-        let root_node_id = taffy::NodeId::from(self.root_element().id);
+        let root_element_id = taffy::NodeId::from(self.root_element().id);
 
         // println!("\n\nRESOLVE LAYOUT\n===========\n");
 
-        taffy::compute_root_layout(self, root_node_id, available_space);
-        taffy::round_layout(self, root_node_id);
+        taffy::compute_root_layout(self, root_element_id, available_space);
+        taffy::round_layout(self, root_element_id);
+
+        // Root element's size is viewport size
+        let root_element = self.get_node_mut(self.root_element().id).unwrap();
+
+        root_element.final_layout.size = taffy::Size {
+            width: size.width.to_f32_px(),
+            height: size.height.to_f32_px(),
+        };
+
+        root_element.clamp_scroll_offset();
```

Comment:
I believe that the way the web works is that there is an implicit "viewport" node that lives above the HTML element that has a fixed size, with the HTML element itself being a normal node. This is probably close enough for the time being, although I don't like overriding the layout size (how does this work when generating "full height" screenshots?)

Reply:
It would be good to have a way to leave the viewport size undefined (determined by layout) as this is used to render "full height screenshots" in the screenshot example (compare output of `cargo run --release --example screenshot https://servo.org` on this branch vs. main)

Reply:
Ok I've kind of restored the old behaviour where we keep a scroll property on the document, which gets bubbled up to from the root node. That seems to be the simplest way to avoid this issue for now, as I don't think we need to go so far as to actually have a viewport node in the tree yet.